sympy/solvers/solveset.py

```diff
@@ -366,26 +390,14 @@ def _solve_as_rational(f, symbol, domain):
 def _solve_trig(f, symbol, domain):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
-    f_original = f
+    f_orig = f
     f = f.rewrite(exp)
-    f = together(f)
-    g, h = fraction(f)
-    y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        result = Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-        return Intersection(result, domain)
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
+    soln = _solveset(f, symbol, S.Complexes)
+    if isinstance(soln, ConditionSet):
```

Comment:
Previously `conditionSet ` come with` exp` form of the trig equation. Now if solveset can't solve using `rewrite(exp)` then try to solve without converting that into `exp` .
And it can't solve then returns `ConditionSet` in `trigonometricFunction`.